packages/plugin/src/index.js

```diff
@@ -1,17 +1,48 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { VIS_TYPE_PIVOT_TABLE } from '@dhis2/analytics'
+import { useDataEngine } from '@dhis2/app-runtime'
 
+import { apiFetchLegendSet } from './api/legendSets'
 import ChartPlugin from './ChartPlugin'
 import PivotPlugin from './PivotPlugin'
 
 const VisualizationPlugin = props => {
+    const engine = useDataEngine()
+    const [legendSet, setLegendSet] = useState(null)
+
+    const hasLegendSet =
+        props.visualization.legendSet && props.visualization.legendSet.id
+
+    useEffect(() => {
+        const fetchLegendSet = async engine => {
+            if (hasLegendSet) {
```

Comment:
I put this in my sample code but I think we should probably be explicit and re-do the conditional here - otherwise there's an implicit dependency on `hasLegendSet`.  It *should* be fine for now since `hasLegendSet` is just a wrapper around on `props.visualization.legendSet` which is a dependency of the `useEffect`, but it would be easy for us to forget about this and break the effect later.  So I think we should probably just keep this as `props.visualization.legendSet && props.visualization.legendSet.id`, or add `hasLegendSet` to the dep list.

Reply:
ok, I'll revert.